### ISSUE TITLE
[DOC]Search templates loop works wrong

### ISSUE THREAD

**Issue by SonaGharibyan** [opensearch-project/documentation-website#5054]
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [ ] Add new documentation
- [x] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Having issues with using search templates loops [https://opensearch.org/docs/latest/search-plugins/search-template/#loops](url)
I have this query

```json
{
   "source":{
    "query":{
      "bool":{
         "filter":[
            {
               "terms":{
                  "field":["{{#names}}","{{.}}","{{/names}}","3"]
               }
            }
         ]
      }
   }
},
   "params":{
      "names":["1", "2"]
   }
}
```
which should return items where "field" equals to "1", "2" or "3". But it also return items where "field" is to empty string ("").

Is there another way to destruct "names" array?
pls update docs accordingly 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
Found similar question just from Elastic Search  -> [https://stackoverflow.com/questions/67590223/how-to-pass-array-to-elasticsearch-search-template-using-mustache](url)

**Comment by kolchfa-aws** [comment]
@SonaGharibyan Can you try passing 1 and 2 as strings:
`"params":{
      "names":["1", "2"]
   }`
And also passing the whole query as a string in the "source" parameter. (Note the format in [this section](https://opensearch.org/docs/latest/search-plugins/search-template/#loops)).

**Comment by SonaGharibyan** [comment]
@kolchfa-aws Yes sorry regarding "params" it was a typo I've updated it. And I've also tried to pass "source" as a string parameter and it also returns all the values where "team" filed (in this example) is empty string ("").
```json
{
  "source": "{\"query\":{\"terms\":{\"team\":[\"{{#myTeams}}\",\"{{.}}\",\"{{/myTeams}}\"]}}}",
  "params": {
    "myTeams":  ["Team A", "Team C"]
  }
}
```

**Comment by kolchfa-aws** [comment]
@SonaGharibyan Mustache inserts a comma after every array element so you're getting a match on a phantom empty element produced by a comma after the last term. To avoid this problem, run the query as follows:
`GET _search/template
{
  "source": "{\"query\":{\"terms\":{\"team\":{{#toJson}}myTeams{{/toJson}}}}}",
  "params": {
    "myTeams":  ["Team A", "Team C"]
  }
}`

**Comment by SonaGharibyan** [comment]
@kolchfa-aws but I also need to add extra elements like "3" here
```json
{
   "source":{
    "query":{
      "bool":{
         "filter":[
            {
               "terms":{
                  "field":["{{#names}}","{{.}}","{{/names}}","3"]
               }
            }
         ]
      }
   }
},
   "params":{
      "names":["1", "2"]
   }
}
```

**Comment by kolchfa-aws** [comment]
Can you just include "3" as an element of the `names` array?

**Comment by SonaGharibyan** [comment]
@kolchfa-aws no, in my case this two data are coming from different places and I don't have access to names array.

**Comment by kolchfa-aws** [comment]
@opensearch-project/opensearch-core Could someone take a look at this issue please? Thanks!

**Comment by natebower** [comment]
Closing as stale
